storage/schemadb/src/lib.rs

```diff
@@ -259,29 +265,42 @@ impl DB {
 
     /// Writes a group of records wrapped in a [`SchemaBatch`].
     pub fn write_schemas(&self, batch: SchemaBatch) -> Result<()> {
+        let mut put_sizes: HashMap<ColumnFamilyName, usize> = HashMap::new();
+        let mut delete_counts: HashMap<ColumnFamilyName, usize> = HashMap::new();
+
         let db_batch = rocksdb::WriteBatch::new();
-        for (cf_name, key, write_op) in &batch.rows {
+        for (cf_name, rows) in batch.rows {
             let cf_handle = self.get_cf_handle(cf_name)?;
-            match write_op {
-                WriteOp::Value(value) => db_batch.put_cf(cf_handle, &key, &value),
-                WriteOp::Deletion => db_batch.delete_cf(cf_handle, &key),
+            for (key, write_op) in rows {
+                match write_op {
+                    WriteOp::Value(value) => {
+                        let size = put_sizes.entry(&cf_name).or_insert(0);
+                        (*size) += key.len() + value.len();
+
+                        db_batch.put_cf(cf_handle, &key, &value)
```

Comment:
Actually, a concrete suggestion for improvement - put the .map_err? here directly, then increment the proper counter or observe the value afterwards.  You'll have to do it for both match cases but the you can get rid of the hash maps as well as the code below.

Reply:
Spiritual discussions are welcome as well @zamsden 💕

But the transaction commits via `self.inner.write_opt(&db_batch)` below, we need to do the counter bumps after that line.

Originally we loop through batch.rows again after the commit, but now that the loop becomes nested, I felt it might be cleaner like what I do now. (plus I thought consuming `batch.row` instead of looping through references had eliminated the copying of the values -- naive, `db_batch` takes only references.)

But yeah It can save the mem allocations if we loop again below. Let me change it back.